### PR TITLE
chore: update storybook

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
- Update Storybook and related dependencies
- Delete the stories for old widget components
- Rewrite stories for new widgets(mdx to tsx)

Note: local chromatic doesn't work because of this https://github.com/storybookjs/storybook/issues/22531 

## Summary by CodeRabbit

- **New Features**
	- Introduced new component stories across various packages to enhance design system documentation and user experience.

- **Documentation**
	- Updated `.gitignore` files to optimize version control settings for Storybook and build logs.

- **Refactor**
	- Modified UI element positioning in Storybook theming for improved layout management.

- **Style**
	- Adjusted CSS properties for the `Select` component to ensure better visual presentation.

- **Chores**
	- Updated dependencies and scripts in package configurations to maintain up-to-date development tools.

### CODE REVIEW

File: app/client/.gitignore [hunk 1]
```diff
@@ -46,9 +46,6 @@ results/
 /docker/nginx.conf
 /docker/nginx-root.conf
 
-storybook-static/*
```
Comment (KelvinOm) [inline app/client/.gitignore:49]: move this to local package .gitignore

File: app/client/packages/design-system/headless/src/components/Menu/stories/Menu.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * Item props are not pulled up in the ArgsTable, the data can be found [here](https://react-spectrum.adobe.com/react-aria/Menu.html#item).
+ */
+
+const meta: Meta<typeof Menu> = {
+  component: Menu,
+  title: "Design-system/headless/Menu",
+  subcomponents: {
+    //@ts-expect-error: don't need props to pass here
+    MenuList,
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Menu/stories/Menu.stories.tsx:16]: this creates argsTable for subcomponents cc @jsartisan

File: app/client/packages/design-system/headless/src/components/Popover/stories/Popover.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  PopoverTrigger,
+  PopoverContent,
+  PopoverModalContent,
+  Button,
+} from "@design-system/headless";
+import { ControlledPopover as ControlledPopoverExample } from "./ControlledPopover";
+
+/**
+ * A popover is an interactive mini-dialog floating element that displays information related to an anchor element when the element is clicked.
+ *
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Popover/stories/Popover.stories.tsx:14]: Comments shows at doc page as well. cc @jsartisan

File: app/client/src/ce/pages/Editor/Explorer/hooks.tsx [hunk 4]
```diff
@@ -130,17 +130,15 @@ export const useAppWideAndOtherDatasource = () => {
...
+        ds1.name?.toLowerCase()?.localeCompare(ds2.name?.toLowerCase()),
       ),
     [datasourcesUsedInApplication],
   );
   const otherDS = useMemo(
     () =>
-      [...otherDatasourceInWorkspace].sort(
-        (ds1, ds2) =>
-          ds1.name?.toLowerCase()?.localeCompare(ds2.name?.toLowerCase()),
+      [...otherDatasourceInWorkspace].sort((ds1, ds2) =>
```
Comment (KelvinOm) [inline app/client/src/ce/pages/Editor/Explorer/hooks.tsx:140]: Prettier also updated :(

**Review by ichik — DISMISSED** [review]

**Review by jsartisan — DISMISSED** [review]

**Review by ichik — APPROVED** [review]

**Pull request merged**